core/src/main/java/io/ddf/etl/TransformationHandler.java

```diff
@@ -12,70 +12,106 @@
+  @Override
+  public DDF transformScaleMinMax() throws DDFException {
+    return this.transformScaleMinMax(new ArrayList<>(), Boolean.FALSE);
+  }
 
+  @Override
+  public DDF transformScaleMinMax(List<String> columns, Boolean inPlace) throws DDFException {
+    Summary[] summaryArr = this.getDDF().getSummary();
     // Compose a transformation query
-    StringBuffer sqlCmdBuffer = new StringBuffer("SELECT ");
+    StringBuilder sqlCmdBuffer = new StringBuilder("SELECT ");
 
-    for (int i = 0; i < columns.size(); i++) {
-      Column col = columns.get(i);
-      if (!col.isNumeric() || col.getColumnClass() == ColumnClass.FACTOR) {
-        sqlCmdBuffer.append(col.getName()).append(" ");
+    for (int colIndex = 0; colIndex < this.getDDF().getSchema().getNumColumns(); colIndex++) {
+      Column col = this.getDDF().getSchema().getColumn(colIndex);
+
+      if (!isColumnEligibleToScale(col, columns)) {
+        sqlCmdBuffer.append(col.getName()).append(",");
       } else {
-        // subtract min, divide by (max - min)
-        sqlCmdBuffer.append(String.format("((%s - %s) / %s) as %s ", col.getName(), summaryArr[i].min(),
-            (summaryArr[i].max() - summaryArr[i].min()), col.getName()));
+        if (summaryArr[colIndex] == null) {
+          LOG.warn("Missing column summary. transformScaleMinMax ignored for column: " + col.getName());
+
+          sqlCmdBuffer.append(col.getName()).append(" ");
+        } else if (Double.compare(summaryArr[colIndex].max(), summaryArr[colIndex].min()) == 0) {
+          LOG.warn("max equals min. transformScaleMinMax ignored for column: " + col.getName());
+
+          sqlCmdBuffer.append(col.getName()).append(" ");
+        } else {
+          // subtract min, divide by (max - min)
+          sqlCmdBuffer.append(String.format("((%s - %s) / %s) as %s ", col.getName(), summaryArr[colIndex].min(),
+                  (summaryArr[colIndex].max() - summaryArr[colIndex].min()), col.getName()));
+        }
+
+        sqlCmdBuffer.append(",");
       }
-      sqlCmdBuffer.append(",");
     }
     sqlCmdBuffer.setLength(sqlCmdBuffer.length() - 1);
-    sqlCmdBuffer.append("FROM ").append(this.getDDF().getTableName());
+    sqlCmdBuffer.append(" FROM ").append(this.getDDF().getTableName());
 
     DDF newddf = this.getManager().sql2ddf(sqlCmdBuffer.toString(), this.getEngine());
     newddf.getMetaDataHandler().copyFactor(this.getDDF());
-    return newddf;
+
+    return inPlace ? this.getDDF().updateInplace(newddf) : newddf;
   }
 
-  @Override public DDF transformScaleStandard() throws DDFException {
-    Summary[] summaryArr = this.getDDF().getSummary();
-    List<Column> columns = this.getDDF().getSchema().getColumns();
+  @Override
+  public DDF transformScaleStandard() throws DDFException {
+    return this.transformScaleStandard(new ArrayList<>(), Boolean.FALSE);
+  }
 
+  @Override
+  public DDF transformScaleStandard(List<String> columns, Boolean inPlace) throws DDFException {
```

Comment:
yes, I have a ticket to add custom range scaling, will do refactoring then. 
https://adatao.atlassian.net/browse/PE-1872